dsymbol/src/dsymbol/ufcs.d

```diff
@@ -43,23 +49,42 @@ enum string[string] INTEGER_PROMOTIONS = [
 
 enum MAX_NUMBER_OF_MATCHING_RUNS = 50;
 
-private const(DSymbol)* deduceSymbolType(const(DSymbol)* symbol)
+private DeducedSymbolTypeResult deduceSymbolTypeByToken(Scope* completionScope, scope ref const(Token) significantToken, size_t cursorPosition)
 {
+    DeducedSymbolTypeResult result;
+    //Literal type deduction
+    if (significantToken.type is tok!"stringLiteral"){
+        result.deducedSymbolType = completionScope.getFirstSymbolByNameAndCursor(istring("string"), cursorPosition);
```

Comment:
there is a helper in `names.d` that can do translation from "stringLiteral", "intLiteral", etc. to "string", "int", etc. (`symbolNameToTypeName`) - should probably be used here for good practice